packages/grid/_modules_/grid/models/colDef/gridDefaultColumnTypes.ts

```diff
@@ -10,6 +11,7 @@ export const getGridDefaultColumnTypes: () => GridColumnTypesRecord = () => {
     number: { ...GRID_NUMERIC_COL_DEF },
     date: { ...GRID_DATE_COL_DEF },
     dateTime: { ...GRID_DATETIME_COL_DEF },
+    boolean: { ...GRID_BOOLEAN_COL_DEF },
```

Comment:
As a side note, all these props cloning looks wrong. Data grid is never supposed to mutate the props provided by the developer. At best, it's work we don't need to do, at worse, we hide a leak.

```suggestion
    boolean: GRID_BOOLEAN_COL_DEF,
```